index.js

```diff
@@ -189,9 +189,12 @@ function invalidExtras(data, msg) {
   throw new Error(msg.replace("%j", json));
 }
 
-function recordName(record) {
+function recordName(RecordType) {
   /* eslint no-underscore-dangle: 0 */
-  return record._name || record.constructor.name || 'Record';
+  if (RecordType.name && RecordType.name !== 'Record') {
+    return RecordType.name;
+  }
+  return (new RecordType({}))._name || 'Record';
```

Comment:
Is it possible to read the `_name` property off RecordType.prototype? I've not looked at the immutable-js code for a while, so I forget why this function looks in the places it does.